resources/js/script.js

```diff
@@ -26,145 +28,173 @@ $(".add-to-cart").click(function(event){
+	var name = $(this).attr("data-name");
+	shoppingCart.removeItemFromCart(name);
+	displayCart();
+});
+
+
 
 /***************************************/
 // Shopping cart functions
 /* Scope - Determins where a variable or a value is visible */ 
 
-var cart = [];
-//when you use the sintax below, it actually generates an object
+var shoppingCart = {};
+shoppingCart.cart = [];
 
-var Item = function(name, price, count) {
+shoppingCart.Item = function(name, price, count) {
 	this.name = name
 	this.price = price
 	this.count = count
 };
 
-/*
-// new item for shopping cart
-var apple = new Item("Apple", 1.99, 1); // {name, price, count}
-
-cart.push(new Item("Brush", 2.13, 1));
-cart.push(apple);*/
-
-// addItemToCart(name, price, count)
-
-function addItemToCart(name, price, count) {
-	for (var i in cart) {
-		if (cart[i].name === name) {
-			cart[i].count += count;
-			saveCart();
+shoppingCart.addItemToCart = function(name, price, count) {
+	for (var i in this.cart) {
+		if (this.cart[i].name === name) {
+			this.cart[i].count += count;
+			this.saveCart();
 			return;
 		}
 	}
-	var item = new Item(name, price, count);
-	cart.push(item);
-	saveCart();
+	var item = new this.Item(name, price, count);
+	this.cart.push(item);
+	this.saveCart();
 };
 
-// removeItemFromCart(name) // Removes one item
-
-function removeItemFromCart(name) {
-	for (var i in  cart) {
-		if (cart[i].name === name) {
-			cart[i].count --;
-			if (cart[i].count === 0) {
-				cart.splice(i, 1); //it removes "1" item from the array
+shoppingCart.removeItemFromCart = function(name) {
+	for (var i in  this.cart) {
+		if (this.cart[i].name === name) {
+			this.cart[i].count --;
+			if (this.cart[i].count === 0) {
+				this.cart.splice(i, 1); //it removes "1" item from the array
 			}
-			return;
+			break;
 		}
 	}
-	saveCart();
-}
-
-// removeItemFromCartAll(name) //Removes all items with name
+	this.saveCart();
+};
 
-function removeItemFromCartAll(name) {
-	for (var i in cart) {
-		if (cart[i].name === name) {
-			cart.splice(i, 1);
+shoppingCart.removeItemFromCartAll = function(name) {
+	for (var i in this.cart) {
+		if (this.cart[i].name === name) {
+			this.cart.splice(i, 1);
 			break;
 		}
 	}
-	saveCart();
-}
-
-
-//clearCart()
-
-function clearCart() {
-	cart = [];
-	saveCart();
-}
+	this.saveCart();
+};
 
-//countCart() -> Returns total count
+shoppingCart.clearCart = function() {
+	this.cart = [];
+	this.saveCart();
+};
 
-function countCart() {
+shoppingCart.countCart = function() {
 	var totalCount = 0;
-	for (var i in cart){
-		totalCount += cart[i].count;
+	for (var i in this.cart){
+		totalCount += this.cart[i].count;
 	}
 	return totalCount;
-}
-
-//totalCart() -> Returns the total cost
+};
 
-function totalCart() {
+shoppingCart.totalCart = function() {
 	var totalCost = 0;
-	for (var i in cart) {
-		totalCost += cart[i].price * cart[i].count;
+	for (var i in this.cart) {
+		totalCost += this.cart[i].price * this.cart[i].count;
 	}
```

Comment:
there must be a shorter way to calculate the total cost without using a for, or?

Reply:
I will try and find the solution. :)